app/models/user/permissions.rb

```diff
@@ -12,6 +12,18 @@ module User::Permissions
     alias_attribute :allowed_service_ids, :member_permission_service_ids
   end
 
+  def permission_for_event?(event)
+    case event
+    when Cinstances::CinstancePlanChangedEvent, Cinstances::CinstanceCancellationEvent,
+           Cinstances::CinstanceExpiredTrialEvent, Applications::ApplicationCreatedEvent
+      has_permission?(:plans) && has_permission?(:partners) && has_access_to_service?(event.service.id)
+    when Services::ServiceDeletedEvent
+      has_access_to_service?(event.service_id)
```

Comment:
Using `event.service.id` in line 19 and `event.service_id` here. Intentional or they could be both `event.service_id`?

Reply:
@guicassolato Intentional. It is what we save in each kind of event. If I did `event.service_id` in both, it would mean that I would have to change all classes in the previous when to contain `service_id` besides the `service` that they already contain.
An example of this:
https://github.com/3scale/porta/blob/76daf670e2d92c7253b2d191c05dcaf7c7f6368c/app/events/services/service_deleted_event.rb#L8
https://github.com/3scale/porta/blob/76daf670e2d92c7253b2d191c05dcaf7c7f6368c/app/events/cinstances/cinstance_plan_changed_event.rb#L5